crates/ruff_linter/src/docstrings/mod.rs

```diff
@@ -51,13 +87,16 @@ pub(crate) struct DocstringBody<'a> {
 
 impl<'a> DocstringBody<'a> {
     pub(crate) fn as_str(self) -> &'a str {
-        &self.docstring.contents[self.docstring.body_range]
+        &self.docstring.source[self.range()]
     }
 }
 
 impl Ranged for DocstringBody<'_> {
     fn range(&self) -> TextRange {
-        self.docstring.body_range + self.docstring.start()
+        TextRange::new(
+            self.docstring.start() + self.docstring.flags().opener_len(),
+            self.docstring.end() - self.docstring.flags().closer_len(),
+        )
```

Comment:
Nit: It would be nice if we could avoid copying this computation. Consider adding a `content_range` to `StringLiteral` similar to https://github.com/astral-sh/ruff/blob/e1781485257783d4b714a9a33c6f8fed4f1477fb/crates/ruff_python_ast/src/expression.rs#L221-L228

Reply:
Hehe. I was planning this for a followup, since there are a number of other places where this method would be useful ;) But I'll add it in this PR, and switch other places in the codebase to this new method as the followup!